blocks/rich-text/index.js

```diff
@@ -54,10 +55,52 @@ export function createTinyMCEElement( type, props, ...children ) {
 	);
 }
 
-export function isLinkBoundary( fragment ) {
-	return fragment.childNodes && fragment.childNodes.length === 1 &&
-		fragment.childNodes[ 0 ].nodeName === 'A' && fragment.childNodes[ 0 ].text.length === 1 &&
-		fragment.childNodes[ 0 ].text[ 0 ] === '\uFEFF';
+/**
+ * Returns true if the node is the inline node boundary. This is used in node
+ * filtering prevent the inline boundary from being included in the split which
+ * occurs while within but at the end of an inline node, since TinyMCE includes
+ * a placeholder caret character at the end.
+ *
+ * @see https://github.com/tinymce/tinymce/blob/master/src/plugins/link/main/ts/core/Utils.ts
+ *
+ * @param {Node} node Node to test.
+ *
+ * @return {boolean} Whether node is inline boundary.
+ */
+export function isEmptyInlineBoundary( node ) {
+	const text = node.nodeName === 'A' ? node.innerText : node.textContent;
```

Comment:
This is not just on links anymore right? Could happen for bold etc. as well?

Reply:
Correct, in fact, this fixes the previous issue of only accounting for anchor. We test any kind of node, but from what I've tested and assuming from [the link plugin Utils.js](https://github.com/tinymce/tinymce/blob/a4add2940715dd27ef7be8ef42065a8292038787/src/plugins/link/main/ts/core/Utils.ts#L52-L53) is that it uses a different property to test (`innerText` for links, `textContent` for all others).